src/handshake/client.rs

```diff
@@ -90,31 +103,45 @@ impl<S: Read + Write> HandshakeRole for ClientHandshake<S> {
                 result,
                 tail,
             } => {
-                self.verify_data.verify_response(&result)?;
+                let mut config = self.config.take().unwrap();
+
+                self.verify_data.verify_response(&result, &mut config)?;
                 debug!("Client handshake done.");
-                let websocket =
-                    WebSocket::from_partially_read(stream, tail, Role::Client, self.config);
+                let websocket = WebSocket::from_partially_read(stream, tail, Role::Client, config);
                 ProcessingResult::Done((websocket, result))
             }
         })
     }
 }
 
 /// Generate client request.
-fn generate_request(request: Request, key: &str) -> Result<Vec<u8>> {
+fn generate_request<Ext>(
+    request: Request,
+    key: &str,
+    config: &mut Option<WebSocketConfig<Ext>>,
+) -> Result<Vec<u8>>
+where
+    Ext: WebSocketExtension,
+{
+    let request = match config {
+        Some(ref mut config) => config.encoder.on_make_request(request),
+        None => request,
+    };
```

Comment:
Could it be rewritten like this?

`let request = config.map(|config| config.encoder.on_make_request(request)).unwrap_or(request);`

It would probably look cleaner like this.